src/vanilla/Model/ParameterTransformations.cs

```diff
@@ -13,6 +13,10 @@ namespace AutoRest.TypeScript.Model
     /// </summary>
     public class ParameterTransformations
     {
+        /// <summary>
+        /// The names of input parameters that are in the options parameter.
+        /// </summary>
+        private readonly ISet<string> inputParametersInOptions;
```

Comment:
Does the parameter itself not know this based on `.Required`?

Reply:
`.Required` doesn't quite cover whether or not a parameter is included in the `options` parameter. [This function](https://github.com/Azure/autorest.typescript/pull/211/files#diff-a0945a4f62f5a6e958b85ad35db7086eR756) was added to share the logic for what determines whether a parameter is in `options` or not.

Reply:
Makes sense--things seem OK as is, then.